src/Firestore/WriteBatch.php

```diff
@@ -0,0 +1,547 @@
+     * // using {@see Google\Cloud\Firestore\FieldPath} will properly escape each element.
+     *
+     * use Google\Cloud\Firestore\FieldPath;
+     *
+     * $batch->updatePaths([
+     *     ['path' => new FieldPath(['cryptoCurrencies', 'big$$$coin']), 'value' => 5.51]
+     * ]);
+     * ```
+     *
+     * @param string $documentName The document name.
+     * @param array[] $data A list of arrays of form `[FieldPath|string $path, mixed $value]`.
+     * @param array $options Configuration options
+     * @return WriteBatch
+     */
+    public function updatePaths($documentName, array $data, array $options = [])
+    {
+        if ($this->isAssoc($data)) {
+            throw new \InvalidArgumentException(
+                'Field data must be provided as a list of arrays of form `[string|FieldPath $path, mixed $value]`.'
+            );
+        }
+
+        $paths = [];
+        $values = [];
+        foreach ($data as $field) {
+            $this->arrayHasKeys($field, ['path', 'value']);
+
+            $paths[] = ($field['path'] instanceof FieldPath)
+                ? $field['path']
+                : FieldPath::fromString($field['path']);
+
+            $values[] = $field['value'];
+        }
+
+        $fields = $this->valueMapper->buildDocumentFromPathsAndValues($paths, $values);
+
+        return $this->update($documentName, $fields, $options);
+    }
+
+    /**
+     * Delete a Firestore document.
+     *
+     * Example:
+     * ```
+     * $batch->delete($documentName);
+     * ```
+     *
+     * @codingStandardsIgnoreStart
+     * @param string $documentName The document to delete.
+     * @param array $options Configuration Options
+     * @return WriteBatch
+     * @codingStandardsIgnoreEnd
+     */
+    public function delete($documentName, array $options = [])
+    {
+        $this->writes[] = $this->createDatabaseWrite(self::TYPE_DELETE, $documentName, $options);
+
+        return $this;
+    }
+
+    /**
+     * Enqueue a Transform operation.
+     *
+     * Note that UPDATE cannot follow TRANSFORM, so once any transforms are
+     * enqueued (either by this method, or by using the `FirestoreClient::SERVER_TIMESTAMP`
+     * sentinel value), any subsequent updates will fail.
+     *
+     * Example:
+     * ```
+     * $batch->transform($documentName, [
+     *     [
+     *         'fieldPath' => 'lastLoginTime',
+     *         'setToServerValue' => WriteBatch::REQUEST_TIME
+     *     ]
+     * ]);
+     * ```
+     *
+     * @param string $documentName The document to apply the transformation to.
+     * @param array[] $transforms {
+     *     A list of Document transformations.
+     *
+     *     @param string $fieldPath The path of the field.
+     *     @param string $setToServerValue Sets the field to the given server value.
+     * }
+     * @param array $options Configuration options.
+     * @return WriteBatch
+     */
+    private function transform($documentName, array $transforms = [], array $options = [])
```

Comment:
Is this intended to be public?

Reply:
nope

Reply:
Ah okay :). Can we drop the snippets and move this down below the rest of the public methods then?